src/compiler/scanner.ts

```diff
@@ -998,6 +1033,25 @@ module ts {
                     return token = SyntaxKind.EndOfFileToken;
                 }
                 let ch = text.charCodeAt(pos);
+
+                if (inJSXChild) {
+                    switch (ch) {
+                        case CharacterCodes.closeBrace:

```

Comment:
I'm sorry but I don't understand the direction you're pointing me here.
When in jsx child, the normal scanning process is *shunted* and the only token handled are `{`,`}`,`<` and `</`  (we might add `>` for coherence). Everything else is put in a single JSXText token.
Would you prefer than I just break the switch for open and close brace and let the normal process take place ?